lcviewernoqt/painters/opengl/sources/cacher.cpp

```diff
@@ -0,0 +1,331 @@
+#include "cacher.h"
+using namespace lc::viewer::opengl;
+
+Cacher::Cacher()
+{
+  _cacher_dashes_sum=0; 
+  _cacher_dashes_size=0;
+  _model=glm::mat4(1.0f);
+  readyFreshPack();
+}
+
+Cacher::~Cacher()
+{
+}
+
+void Cacher::setShaderBook(struct Shaders_book& book)
+{
+  _shaders=book;
+}
+
+void Cacher::setFontBook(Font_Book& book)
+{
+  _fonts=book;
+}
+
+//---------------------------For Matrix/ Vectors/ Coordinate-----------
+void Cacher::updateScale(float scale)
+{
+  _model=glm::scale(_model,glm::vec3(scale,scale,scale));
+}
+
+void Cacher::updateTranslate(float x,float y)
+{
+  _model=glm::translate(_model,glm::vec3(x,y,0.0));
+}
+
+void Cacher::updateRotate(float angle)
+{
+  _model=glm::rotate(_model, angle, glm::vec3(0.0f ,0.0f ,1.0f) );
+}
+
+void Cacher::resetTransformations()
+{
+  _model=glm::mat4(1.0f);
+}
+
+double Cacher::getScale()
+{
+  return _model[2][2];
+}
+
+double Cacher::getTranslateX()
+{
+  return _model[3][0];
+}
+
+double Cacher::getTranslateY()
+{
+  return _model[3][1];
+}   
+
+//------------------------------------for properties ( painter calls)------------------------------------
+void Cacher::addVertex(float x,float y,float z)
+{
+  if(_current_vertices.size()==0)
+    _path_distance=0.0f;
+  else
+  {
+    glm::vec2 P=glm::vec2( (*(_current_vertices.rbegin())) );
+    glm::vec2 Q=glm::vec2( x , -y);
+    float d=glm::length(P-Q);
+    _path_distance+=d;
+  }
+  _current_vertices.push_back( glm::vec4(x,-y,z,_path_distance) );
+}
+
+void Cacher::appendVertexData()
```

Comment:
Exact code duplication with `Renderer::appendVertexData()`
Renaming variables to make CodeFactor check passes doesn't remove the code duplication

Reply:
Done!!